src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/ApproveVacancyReviewCommandHandler.cs

```diff
@@ -101,13 +103,27 @@ private void Validate(VacancyReview review)
             }
         }
 
-        private bool GetHasVacancyBeenTransferredSinceReviewWasCreated(VacancyReview review, Vacancy vacancy)
+        private bool HasVacancyBeenTransferredSinceReviewWasCreated(VacancyReview review, Vacancy vacancy)
+        {
+            return review.VacancySnapshot.TransferInfo == null && vacancy.TransferInfo != null;
+        }
+
+        private async Task<bool> HasProviderBeenBlockedSinceReviewWasCreatedAsync(Vacancy vacancy)
+        {
```

Comment:
This looks fine, my only slight concern is that if the provider is blocked/unblocked quickly but the source vacancy has been transferred, we are not looking at the source vacancy transfer info reason or state to work out the closure reason.

Reply:
For transferred vacancies we do look at the vacancy TransferInfo to determine the closure reason.  See `HasVacancyBeenTransferredSinceReviewWasCreated`

This handles non-transferred vacancies (Employer vacancies) for providers that have been blocked since the review was created. I think this is the only way we can determine this.